pyqtorch/modules/circuit.py

```diff
@@ -40,6 +40,22 @@ def __init__(self, n_qubits: int, operations: list):
         self.n_qubits = n_qubits
         self.operations = torch.nn.ModuleList(operations)
 
+    def __mul__(self, other: QuantumCircuit) -> QuantumCircuit:
+        if self.n_qubits != other.n_qubits:
+            raise ValueError(
+                (
+                    f"Number of Qubits don't match. "
+                    f"Left gate is applied on a {self.n_qubits} qubit system and "
+                    f"right gate is applied on a {other.n_qubits} qubit system."
+                )
+            )
+
+        if isinstance(other, QuantumCircuit):
+            return QuantumCircuit(self.n_qubits, other.operations.append(self.operations))
```

Comment:
Is the order correct here? I would expect that doing `circuit1 * circuit2` would stick the second circuit behind the first. This is the other way round, right?

Reply:
I used the moduleList in a first-in, first-applied way. However, I realized now that I messed up the matrices method, which works the other way around. Oops.